pkgs/by-name/py/pyfa/package.nix

```
@@ -1,28 +1,114 @@
 {
   lib,
-  appimageTools,
-  fetchurl,
+  python3Packages,
+  fetchFromGitHub,
+  gsettings-desktop-schemas,
+  adwaita-icon-theme,
+  wrapGAppsHook3,
+  gdk-pixbuf,
 }:
 let
-  version = "2.60.2";
+  version = "2.61.0";
 in
-appimageTools.wrapType2 {
+python3Packages.buildPythonApplication rec {
   inherit version;
   pname = "pyfa";
+  format = "other";
 
-  src = fetchurl {
-    name = "pyfa-appimage-${version}";
-    url = "https://github.com/pyfa-org/Pyfa/releases/download/v${version}/pyfa-v${version}-linux.AppImage";
-    hash = "sha256-6doetQ6E1Occ/SqewfxRqPEX1MnUuFomm+8VmetIz4Y=";
+  src = fetchFromGitHub {
+    owner = "pyfa-org";
+    repo = "Pyfa";
+    rev = "v${version}";
+    hash = "sha256-VSuRQugUODc+LbhKbzsA09pnqPEIAt0pQS2An/p7r9A=";
   };
 
+  build-system = [ python3Packages.setuptools ];
+  propagatedBuildInputs = with python3Packages; [
+    wxpython
+    logbook
+    matplotlib
+    python-dateutil
+    requests
+    sqlalchemy_1_4
+    cryptography
+    markdown2
+    beautifulsoup4
+    pyaml
+    roman
+    numpy
+    python-jose
+    requests-cache
+  ];
+
+  buildInputs = [
+    python3Packages.pyinstaller
+    gsettings-desktop-schemas
+    adwaita-icon-theme
+  ];
+
+  dontWrapGApps = true;
+  nativeBuildInputs = [
+    wrapGAppsHook3
+    gdk-pixbuf
+  ];
+
+  patchPhase = ''
+    cat > setup.py <<EOF
+      from setuptools import setup
+      setup(
+        name = "${pname}",
+        version = "${version}",
+        scripts = ["pyfa.py"],
+        packages = setuptools.find_packages(),
+      )
+    EOF
+  '';
+
+  configurePhase = ''
+    python db_update.py
+  '';
+
+  buildPhase = ''
+    ${python3Packages.pyinstaller}/bin/pyinstaller \
+      --clean \
+      --noconfirm \
+      pyfa.spec
+  '';
```

Fairly sure `pyinstaller` would already be in PATH

```suggestion
  buildPhase = ''
  	runHook preBuild
  	
    pyinstaller --clean --noconfirm pyfa.spec
    
    runHook postBuild
  '';
```